### PR TITLE
Add better tags input

### PR DESCRIPTION
**Pull request by franciclo**
Add autocomplete feature using radix-trie.
Got some validation errors on topic edit after rebase.

### CODE REVIEW

File: lib/api-v2/comments/csv.js [hunk 1]
```diff
@@ -48,7 +48,7 @@ function escapeTxt (text) {
 
 app.get('/comments.csv',
 middlewares.users.restrict,
-middlewares.forums.findByForumId,
+middlewares.forums.findByName,
```
Comment (mjlescano) [inline lib/api-v2/comments/csv.js]: Why this change?

Comment (franciclo) [inline lib/api-v2/comments/csv.js]: Because it really was finding by name not id
